src/content/docs/en/reference/integrations-reference.mdx

```diff
@@ -452,10 +462,143 @@ const integration = {
 }
 ```
 
-### `astro:config:done`
+### `astro:routes:resolved`
+
+<Since v="5.0.0" />
 
 **Previous hook:** [`astro:config:setup`](#astroconfigsetup)
 
+**Next hook:** [`astro:config:done`](#astroconfigdone)
+
+**When:** In `astro dev`, it also runs if a file based route changes (added/removed/updated).
+
+**Why:** To access routes and their metadata
+
+```js
+'astro:routes:resolved'?: (options: {
+  routes: IntegrationResolvedRoute[];
+  logger: AstroIntegrationLogger;
+}) => void | Promise<void>;
+```
+
+#### `routes` option
+
+**Type:** [`IntegrationResolvedRoute[]`](#integrationresolvedroute-type-reference)
+
+A list of all routes alongside their associated metadata.
+
+You can reference the full `IntegrationResolvedRoute` type below, but the most common properties are:
+
+- `entrypoint` - the input file path relative to the project root
+- `pathname` - the output file URL (undefined for routes using `[dynamic]` and `[...spread]` params)
```

Comment:
I'm not a fan of defining twice. If these two are common, is there instead some kind of example with these two? something like:

The following example accesses the `entrypoint` and `pathname` ... 

And don't define them here? Is there some kind of minimal example usage that would work here?

Reply:
FYI I just copied and adapted what was in `astro:build:done` so maybe we can also update it there with you think is best!

Reply:
Gotta say I'm struggling to find a good simple example, the `routes` related logic is usually quite involved :sweat:. @Fryuni do you have an idea?